compiler/Main.hs

```diff
@@ -182,7 +182,17 @@ javaCodegen Java {..} = do
                     let content =
                           if no_banner
                           then code
-                          else (commonHeader "//" bondFile javaFile <> code)
+                          else (commonHeader "//" safeBondFile safeJavaFile <> code)
+                              where
+                                  -- javac will always treat "\u" as the start
+                                  -- of a unicode escape sequence, and will
+                                  -- error out if it isn't followed by a valid
+                                  -- code. This breaks compilation of generated
+                                  -- code if either path has components that
+                                  -- start with u.
+                                  backToForward s = LT.unpack $ LT.replace (LT.pack "\\") (LT.pack "/") (LT.pack s)
```

Comment:
How about `backToForward s = map (\c -> if c == '\\' then '/'; else c) s` so we don't have to do all the packing and packing?

Reply:
Done.